nucleus/utils.py

```diff
@@ -422,3 +434,87 @@ def paginate_generator(
             yield json_value
         if not next_token:
             break
+
+def fetch_image(s3_url: str) -> Image.Image:
+    s3_bucket = s3_url.split("//")[-1].split("/")[0]
+    s3_output = s3_url.split("//")[-1].split(s3_bucket)[-1][1:]
+    image = Image.open(
+        boto3.resource("s3").Bucket(s3_bucket).Object(s3_output).get()["Body"]
+    )
+    return image
+
+def fetch_chip(s3_prefix: str) -> tuple[str, str]:
+    s3_bucket = s3_prefix.split("//")[-1].split("/")[0]
+    s3_output = s3_prefix.split("//")[-1].split(s3_bucket)[-1][1:]
+    try:
+        boto3.resource("s3").Bucket(s3_bucket).Object(s3_output + '.jpeg').load()
+    except:
```

Comment:
You should filter on exception type so this doesn't hide other bugs/failures. At the very least, `except Exception` will still raise `KeyboardInterrupt` so you can still ctrl+c to quit.